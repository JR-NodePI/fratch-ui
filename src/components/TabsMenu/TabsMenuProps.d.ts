import { CSSProperties } from 'react';

import { IconCheck as Icon } from '../Icons';

export type Tab = {
  active?: boolean;
  color?: CSSProperties['color'];
  label?: string;
  Icon?: typeof Icon;
};

export type TabEvent = Pick<Tab, 'label' | 'color' | 'Icon'> & {
  index: number;
};

export type TabsMenuProps = {
  addable?: boolean;
  className?: string;
  editable?: boolean;
  newTabTemplate?: Pick<Tab, 'label' | 'Icon' | 'color'>;
  onTabAdd?: (tabData: TabEvent) => void;
  onTabClick?: (tabData: TabEvent) => void;
  onTabEdit?: (tabData: TabEvent) => void;
  onTabRemove?: (tabData: TabEvent) => void;
  onTabsChange?: (tabs: Tab[]) => void;
  removable?: boolean;
  tabs?: Tab[];
  tabsDefault?: Tab[];
};
