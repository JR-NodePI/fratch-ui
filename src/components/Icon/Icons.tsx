export { default as IconArrowDown } from './IconArrowDown';
export { default as IconClose } from './IconClose';
export { default as IconPlus } from './IconPlus';
export { default as IconCheck } from './IconCheck';
