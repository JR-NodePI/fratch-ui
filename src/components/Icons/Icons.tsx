export { default as IconArrowDown } from './Icons/IconArrowDown';
export { default as IconCheck } from './Icons/IconCheck';
export { default as IconClose } from './Icons/IconClose';
export { default as IconDownload } from './Icons/IconDownload';
export { default as IconEdit } from './Icons/IconEdit';
export { default as IconError } from './Icons/IconError';
export { default as IconExport } from './Icons/IconExport';
export { default as IconImport } from './Icons/IconImport';
export { default as IconInfo } from './Icons/IconInfo';
export { default as IconPause } from './Icons/IconPause';
export { default as IconPin } from './Icons/IconPin';
export { default as IconPlay } from './Icons/IconPlay';
export { default as IconPlus } from './Icons/IconPlus';
export { default as IconSuccess } from './Icons/IconSuccess';
export { default as IconUpload } from './Icons/IconUpload';
export { default as IconVerticalDots } from './Icons/IconVerticalDots';
export { default as IconWarning } from './Icons/IconWarning';
