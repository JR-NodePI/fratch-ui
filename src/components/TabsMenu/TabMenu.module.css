.tab_menu {
  overflow: auto;
  display: flex;
  width: 100%;
  box-sizing: border-box;
  flex-wrap: nowrap;
  padding-bottom: 16px;
}
.tab_menu > ul {
  border-bottom: 1px solid var(--ft-color-grey-1);
  background: var(--ft-color-grey-3);
  margin: 0;
  padding: 0;
  list-style: none;
  flex-wrap: nowrap;
  display: flex;
  gap: 1px;
  padding: 8px 8px 0 8px;
  height: calc(var(--ft-tab-height) + 1px);
  flex-wrap: nowrap;
  width: auto;
  flex: 1 1 100%;
  position: relative;
  z-index: 1;
}
.tab_menu > ul::after {
  content: ' ';
  position: absolute;
  z-index: 2;
  left: 0;
  width: 100%;
  height: 16px;
  top: calc(100% + 1px);
  background-image: linear-gradient(var(--ft-color-white), rgba(0, 0, 0, 0));
}

/***  Tab Item  ***/
.tab_menu .tab_item {
  flex-wrap: nowrap;
  position: relative;
  z-index: 1;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 300px;
  min-width: 100px;
  border-top-left-radius: var(--ft-button-default-border-radius);
  border-top-right-radius: var(--ft-button-default-border-radius);
  background-color: var(--ft-color-white);
  border: 1px solid var(--ft-color-grey-1);
  margin-bottom: -1px;
  text-decoration: none;
}

.tab_menu .tab_item .tab_trigger {
  flex-wrap: nowrap;
  display: flex;
  gap: 6px;
  align-items: center;
  overflow: hidden;
  text-decoration: none;
  cursor: default;
  padding-left: var(--ft-button-small-padding-x);
  height: var(--ft-tab-height);
  width: 100%;
  color: var(--ft-color-text);
  justify-content: space-between;
}

.tab_menu .tab_item .tab_trigger:last-child {
  padding-right: var(--ft-button-small-padding-x);
}

.tab_menu .tab_item .tab_trigger .tab_icon {
  width: 1.3em;
  min-width: 1.3em;
  font-size: 1.3em;
  margin-left: -6px;
}

.tab_menu .tab_item .tab_trigger .tab_label {
  flex: 1 1 100%;
  max-width: 100%;
  text-align: center;
  box-sizing: border-box;
}
.tab_menu .tab_item .tab_closer {
  position: relative;
  min-width: var(--ft-tab-height);
  height: var(--ft-tab-height);
  overflow: hidden;
}

.tab_menu .tab_item.active {
  z-index: 2;
  box-shadow: var(--ft-box-shadow-2);
  border-bottom: 1px solid var(--ft-color-white);
}

/***  Add Tab  ***/
.tab_menu .add_tab {
  width: 32px;
  height: 32px;
  margin-left: 4px;
  align-self: center;
}
.tab_menu .add_tab a {
  text-decoration: none;
  opacity: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 32px;
  color: var(--ft-color-text);
  cursor: default;
  font-weight: 500;
  border-radius: 100%;
  background-color: var(--ft-color-white);
  transition:
    opacity 0.2s ease-in-out,
    background-color 0.2s ease-in-out;
}
.tab_menu .add_tab a:hover {
  opacity: 1;
  background-color: var(--ft-button-hover-bg-color);
}
