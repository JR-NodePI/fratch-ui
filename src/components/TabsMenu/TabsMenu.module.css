.tab_menu {
  overflow-x: auto;
  overflow-y: hidden;
  display: flex;
  width: 100%;
  box-sizing: border-box;
  flex-wrap: nowrap;
  height: calc(var(--ft-tabs-height) + 19px + var(--ft-tabs-padding));
}
.tab_menu > ul {
  border-bottom: 1px solid var(--ft-color-grey-1);
  background: var(--ft-color-grey-4);
  margin: 0;
  padding: 0;
  list-style: none;
  flex-wrap: nowrap;
  display: flex;
  gap: 1px;
  padding: var(--ft-tabs-padding) var(--ft-tabs-padding) 0
    var(--ft-tabs-padding);
  height: calc(var(--ft-tabs-height) + 1px);
  flex-wrap: nowrap;
  width: auto;
  flex: 1 1 100%;
  position: relative;
  z-index: 1;
}
.tab_menu > ul::after {
  content: ' ';
  position: absolute;
  z-index: 2;
  left: 0;
  width: 100%;
  height: 16px;
  top: calc(100% + 1px);
  background-image: linear-gradient(
    var(--ft-color-background),
    90%,
    rgba(255, 255, 255, 0)
  );
}

/***  Tab Item  ***/
.tab_menu .tab_item {
  flex-wrap: nowrap;
  position: relative;
  z-index: 1;
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 300px;
  min-width: 100px;
  border-top-left-radius: var(--ft-button-default-border-radius);
  border-top-right-radius: var(--ft-button-default-border-radius);
  background-color: var(--ft-color-background);
  border: 1px solid var(--ft-color-grey-1);
  margin-bottom: -1px;
  text-decoration: none;
}

.tab_menu .tab_item .tab_trigger {
  flex-wrap: nowrap;
  display: flex;
  gap: 4px;
  align-items: center;
  overflow: hidden;
  text-decoration: none;
  cursor: default;
  padding-left: var(--ft-button-small-padding-x);
  height: var(--ft-tabs-height);
  width: 100%;
  color: var(--ft-color-text);
  justify-content: space-between;
}
.tab_menu .tab_item .tab_trigger:last-child {
  padding-right: var(--ft-button-small-padding-x);
}
.tab_menu .tab_item .tab_trigger .tab_color {
  display: block;
  position: absolute;
  width: 4px;
  left: 4px;
  top: 4px;
  height: 28px;
  border-radius: 4px;
}
.tab_menu .tab_item .tab_trigger .tab_icon {
  min-width: 1.1em;
  min-height: 1.1em;
  font-size: 1.1em;
  margin-right: 4px;
}
.tab_menu .tab_item .tab_trigger .tab_label {
  flex: 1 1 100%;
  max-width: 100%;
  text-align: center;
  box-sizing: border-box;
}
.tab_menu .tab_item .tab_closer {
  position: relative;
  min-width: var(--ft-tabs-height);
  height: var(--ft-tabs-height);
  overflow: hidden;
}

.tab_menu .tab_item.active {
  z-index: 2;
  box-shadow: var(--ft-box-shadow-2);
  border-bottom: 1px solid var(--ft-color-background);
}

/***  Add Tab  ***/
.tab_menu .add_tab {
  width: 32px;
  height: 32px;
  margin-left: 4px;
  align-self: center;
}
.tab_menu .add_tab a {
  text-decoration: none;
  opacity: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 32px;
  color: var(--ft-color-text);
  cursor: default;
  font-weight: 500;
  border-radius: 100%;
  background-color: var(--ft-color-background);
  transition:
    opacity 0.2s ease-in-out,
    background-color 0.2s ease-in-out;
}
.tab_menu .add_tab a:hover {
  opacity: 1;
  background-color: var(--ft-button-hover-bg-color);
}
