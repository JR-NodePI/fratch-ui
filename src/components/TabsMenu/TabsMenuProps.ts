import { CSSProperties } from 'react';

import { IconCheck as Icon } from '../Icons/Icons';

export type Tab = {
  active?: boolean;
  color?: CSSProperties['color'];
  label?: string;
  Icon?: typeof Icon;
};

export type TabsMenuProps = {
  className?: string;
  editable?: boolean;
  removable?: boolean;
  addable?: boolean;
  newTabTemplate?: Pick<Tab, 'label' | 'Icon' | 'color'>;
  onTabAdd?: (tabData: Pick<Tab, 'label'> & { index: number }) => void;
  onTabClick?: (tabData: Pick<Tab, 'label'> & { index: number }) => void;
  onTabEdit?: (tabData: Pick<Tab, 'label'> & { index: number }) => void;
  onTabRemove?: (tabData: { index: number }) => void;
  onTabsChange?: (tabs: Tab[]) => void;
  tabs?: Tab[];
};
