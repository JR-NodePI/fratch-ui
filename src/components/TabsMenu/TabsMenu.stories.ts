import type { Meta, StoryObj } from '@storybook/react';

import IconCheck from '../Icons/IconCheck';
import IconInfo from '../Icons/IconInfo';
import IconSuccess from '../Icons/Icons/IconSuccess';
import TabsMenu from './TabsMenu';

const meta = {
  title: 'Example/Tabs Menu',
  component: TabsMenu,
  argTypes: {
    onTabClick: { action: 'tab clicked' },
    onTabAdd: { action: 'tab added' },
    onTabRemove: { action: 'tab removed' },
    onTabEdit: { action: 'tab edited' },
    onTabsChange: { action: 'tabs changed' },
  },
  tags: ['autodocs'],
} satisfies Meta<typeof TabsMenu>;

export default meta;

type Story = StoryObj<typeof TabsMenu>;

export const Default: Story = {
  args: {
    tabs: [
      {
        Icon: IconCheck,
        label: 'Tab 1',
        active: true,
      },
      {
        Icon: IconInfo,
        label: 'Tab 2 with more text',
      },
      {
        Icon: IconSuccess,
        label: 'Tab 3',
      },
      {
        label: 'Tab 4 with much more text',
      },
    ],
  },
};

export const Editable: Story = {
  args: {
    editable: true,
    tabs: [
      {
        label: 'Tab 1',
        active: true,
      },
      {
        label: 'Tab 2 with more text',
      },
      {
        label: 'Tab 3',
      },
    ],
  },
};

export const EditableWithTemplate: Story = {
  args: {
    editable: true,
    newTabTemplate: {
      label: 'Template tab',
      Icon: IconInfo,
    },
    tabs: [
      {
        label: 'Tab 1',
        active: true,
      },
    ],
  },
};
