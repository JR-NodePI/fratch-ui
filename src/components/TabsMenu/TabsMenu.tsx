import { useEffect, useState } from 'react';

import { c } from '../../helpers/classNameHelpers';
import ButtonCloser from '../ButtonCloser/ButtonCloser';
import TabLabelEditable from './components/TabLabelEditable';
import { type Tab, type TabsMenuProps } from './TabsMenuProps';

import styles from './TabMenu.module.css';

export default function TabsMenu({
  className,
  editable,
  newTabTemplate,
  onTabAdd,
  onTabClick,
  onTabEdit,
  onTabRemove,
  onTabsChange,
  tabs,
}: TabsMenuProps): JSX.Element {
  const [currentTabs, setCurrentTabs] = useState<Tab[]>(tabs ?? []);

  useEffect(() => {
    onTabsChange?.(currentTabs);
  }, [onTabsChange, currentTabs]);

  const handleTabClick = (currentTabindex: number): void => {
    setCurrentTabs(
      currentTabs.map((tab, index) => ({
        ...tab,
        active: index === currentTabindex,
      }))
    );

    const currentTab = !isNaN(currentTabindex)
      ? currentTabs[currentTabindex]
      : null;
    onTabClick?.({ label: currentTab?.label, index: currentTabindex });
  };

  const handleAddClick = (): void => {
    let newTabs = currentTabs;

    newTabs = currentTabs.map(tab => ({
      ...tab,
      active: false,
    }));

    newTabs = [...(newTabs ?? []), { ...(newTabTemplate ?? {}), active: true }];

    setCurrentTabs(newTabs);

    const newTabIndex = currentTabs.length ?? 0;
    onTabAdd?.({ label: newTabTemplate?.label, index: newTabIndex });
  };

  const handleRemoveClick = (currentTabindex: number): void => {
    const currentTab = !isNaN(currentTabindex)
      ? currentTabs[currentTabindex]
      : null;
    let newTabs = currentTabs;

    const mustActivatePreviousTab = currentTab?.active && currentTabindex > 0;

    if (mustActivatePreviousTab) {
      newTabs = newTabs?.map((tab, index) => ({
        ...tab,
        active: index === currentTabindex - 1,
      }));
    }

    const mustActivateNextTab =
      currentTab?.active && currentTabindex < (newTabs?.length ?? 0) - 1;

    if (mustActivateNextTab) {
      newTabs = newTabs?.map((tab, index) => ({
        ...tab,
        active: index === currentTabindex + 1,
      }));
    }

    newTabs = newTabs?.filter((_tab, index) => index !== currentTabindex);

    setCurrentTabs(newTabs);

    onTabRemove?.({
      index: currentTabindex,
    });
  };

  const handleEditChange = (
    currentTabindex: number,
    newLabel: string
  ): void => {
    setCurrentTabs(
      currentTabs.map((tab, index) => {
        if (index === currentTabindex) {
          return {
            ...tab,
            label: newLabel,
          };
        }
        return tab;
      })
    );

    onTabEdit?.({ label: newLabel, index: currentTabindex });
  };

  return (
    <nav className={c(styles.tab_menu, className)}>
      <ul>
        {currentTabs.map(({ label, Icon, active }, index) => (
          <li
            className={c(styles.tab_item, active ? styles.active : '')}
            key={index}
            title={label}
          >
            <a
              href="#"
              className={c(styles.tab_trigger)}
              onClick={event => {
                event.preventDefault();
                handleTabClick(index);
              }}
            >
              {Icon && <Icon className={c(styles.tab_icon)} />}
              <TabLabelEditable
                className={c(styles.tab_label)}
                label={label}
                editable={editable}
                onChange={(newLabel: string) => {
                  handleEditChange(index, newLabel);
                }}
              />
            </a>
            {editable && (
              <ButtonCloser
                title="Remove tag"
                className={c(styles.tab_closer)}
                onClick={() => handleRemoveClick(index)}
              />
            )}
          </li>
        ))}
        {editable && (
          <li className={c(styles.add_tab)}>
            <a
              href="#"
              onClick={event => {
                event.preventDefault();
                handleAddClick();
              }}
              title="Add new tab"
            >
              +
            </a>
          </li>
        )}
      </ul>
    </nav>
  );
}
