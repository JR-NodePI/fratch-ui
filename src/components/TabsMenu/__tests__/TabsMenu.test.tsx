import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it, vi } from 'vitest';

import { IconCheck } from '../../Icons/Icons';
import TabsMenu from '../TabsMenu';
import { type TabsMenuProps } from '../TabsMenuProps';

describe('TabsMenu.tsx', () => {
  const getComponent = (props: TabsMenuProps = {}) => <TabsMenu {...props} />;
  const setup = (props = {}) => render(getComponent(props));

  it('should render properly', () => {
    const { container } = setup({
      tabs: [
        {
          active: true,
          label: 'label-test-1',
          Icon: IconCheck,
        },
        {
          label: 'label-test-2',
        },
        {
          label: 'label-test-3',
        },
      ],
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with editable addable and removable', () => {
    const { container } = setup({
      editable: true,
      addable: true,
      removable: true,
      tabs: [
        {
          label: 'label-test-1',
          active: true,
        },
        {
          label: 'label-test-2',
        },
      ],
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with className', () => {
    const { container } = setup({
      className: 'className-test',
    });
    expect(container).toMatchSnapshot();
  });

  it('should render with tab colors', () => {
    const mockTabs = [
      { label: 'label-1', color: 'Red' },
      { label: 'label-2', color: 'Blue' },
    ];

    const { container } = setup({
      className: 'className-test',
      tabs: mockTabs,
    });
    expect(container).toMatchSnapshot();
  });

  it('should add new tab based on template in addable mode', async () => {
    const { container } = setup({
      addable: true,
      tabs: [],
      newTabTemplate: { label: 'new-tab-label-from-template' },
    });

    await userEvent.click(screen.getByRole('link', { name: /Add new tab/ }));

    expect(container).toMatchSnapshot();
  });

  it('should perform onTabAdd', async () => {
    const onTabAdd = vi.fn();
    const mockTabs = [{ label: 'label-test-1', active: true }];
    setup({
      addable: true,
      onTabAdd,
      newTabTemplate: { label: 'new-tab-label' },
      tabs: mockTabs,
    });

    await userEvent.click(screen.getByRole('link', { name: /Add new tab/ }));

    expect(onTabAdd).toHaveBeenCalledTimes(1);
    expect(onTabAdd).toHaveBeenCalledWith({
      index: 1,
      label: 'new-tab-label',
    });
  });

  it('should perform onTabsChange', async () => {
    const onTabsChange = vi.fn();
    setup({
      addable: true,
      editable: true,
      removable: true,
      onTabsChange,
    });

    await userEvent.click(screen.getByRole('link', { name: /Add new tab/ }));

    expect(onTabsChange).toHaveBeenCalledTimes(1);
    expect(onTabsChange).toHaveBeenCalledWith([{ active: true }]);
  });

  describe('should perform onTabRemove', async () => {
    it('with next tab active', async () => {
      const mockTabs = [
        { label: 'label-test-1' },
        { label: 'label-test-2', active: true },
        { label: 'label-test-3' },
      ];
      const expectedTabIndexRemoved = 1;

      const onTabRemove = vi.fn();
      const onTabsChange = vi.fn();
      setup({
        removable: true,
        tabs: mockTabs,
        onTabRemove,
        onTabsChange,
      });

      await userEvent.click(
        (
          await screen.getAllByRole('button', {
            name: /Remove tab/,
            hidden: true,
          })
        )?.[expectedTabIndexRemoved]
      );

      expect(onTabsChange).toHaveBeenCalledTimes(1);
      expect(onTabsChange).toHaveBeenCalledWith([
        { label: 'label-test-1', active: false },
        { label: 'label-test-3', active: true },
      ]);

      expect(onTabRemove).toHaveBeenCalledTimes(1);
      expect(onTabRemove).toHaveBeenCalledWith({
        Icon: undefined,
        color: undefined,
        index: expectedTabIndexRemoved,
        label: 'label-test-2',
      });
    });

    it('with previous tab active', async () => {
      const mockTabs = [
        { label: 'label-test-1' },
        { label: 'label-test-2' },
        { label: 'label-test-3', active: true },
      ];
      const expectedTabIndexRemoved = 2;

      const onTabRemove = vi.fn();
      const onTabsChange = vi.fn();
      setup({
        removable: true,
        tabs: mockTabs,
        onTabRemove,
        onTabsChange,
      });

      await userEvent.click(
        (
          await screen.getAllByRole('button', {
            name: /Remove tab/,
            hidden: true,
          })
        )?.[expectedTabIndexRemoved]
      );

      expect(onTabsChange).toHaveBeenCalledTimes(1);
      expect(onTabsChange).toHaveBeenCalledWith([
        { label: 'label-test-1', active: false },
        { label: 'label-test-2', active: true },
      ]);

      expect(onTabRemove).toHaveBeenCalledTimes(1);
      expect(onTabRemove).toHaveBeenCalledWith({
        Icon: undefined,
        color: undefined,
        index: expectedTabIndexRemoved,
        label: 'label-test-3',
      });
    });
  });

  it('should perform onTabClick', async () => {
    const onTabClick = vi.fn();
    const mockTabs = [
      { label: 'label-test-1', active: true },
      { label: 'label-test-2' },
      { label: 'label-test-3' },
    ];
    setup({
      tabs: mockTabs,
      onTabClick,
    });

    await userEvent.click(await screen.getByText('label-test-2'));

    expect(onTabClick).toHaveBeenCalledTimes(1);
    expect(onTabClick).toHaveBeenCalledWith({
      index: 1,
      label: 'label-test-2',
    });
  });

  it('should perform onTabEdit', async () => {
    const onTabEdit = vi.fn();
    const mockTabs = [
      { label: 'label-test-1' },
      { label: 'label-test-2', active: true },
      { label: 'label-test-3' },
    ];
    const expectedTabIndexEdited = 1;
    setup({
      editable: true,
      tabs: mockTabs,
      onTabEdit,
    });

    await userEvent.click(
      (
        await screen.getAllByRole('button', {
          name: /Edit tag/,
          hidden: true,
        })
      )?.[expectedTabIndexEdited]
    );

    await userEvent.type(
      screen.getByRole('textbox', { name: /Edit tag/i }),
      '-and-new-value'
    );
    await userEvent.keyboard('{enter}');

    expect(onTabEdit).toHaveBeenCalledTimes(1);
    expect(onTabEdit).toHaveBeenCalledWith({
      index: expectedTabIndexEdited,
      label: 'label-test-2-and-new-value',
    });
  });
});
