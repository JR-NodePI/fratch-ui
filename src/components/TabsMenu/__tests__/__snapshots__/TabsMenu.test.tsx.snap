// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabsMenu > should add new tab based on template in addable mode 1`] = `
<div>
  <nav
    class="_tab_menu_3abddc"
  >
    <ul>
      <li
        class="_tab_item_3abddc _active_3abddc"
        title="new-tab-label-from-template"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                new-tab-label-from-template
              </span>
            </div>
          </div>
        </div>
      </li>
      <li
        class="_add_tab_3abddc"
      >
        <a
          aria-label="Add new tab"
          href="#"
          title="Add new tab"
        >
          <svg
            aria-hidden="true"
            class="_icon_2be8fc _primary_2be8fc"
            fill="none"
            viewBox="-0.5 -0.5 19 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1v16M1 9h16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`TabsMenu > should render properly 1`] = `
<div>
  <nav
    class="_tab_menu_3abddc"
  >
    <ul>
      <li
        class="_tab_item_3abddc _active_3abddc"
        title="label-test-1"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <svg
            class="_icon_2be8fc _primary_2be8fc _tab_icon_3abddc"
            fill="none"
            viewBox="-1 -3 18 18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 5.917 5.724 10.5 15 1.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="3"
            />
          </svg>
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-test-1
              </span>
            </div>
          </div>
        </div>
      </li>
      <li
        class="_tab_item_3abddc"
        title="label-test-2"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-test-2
              </span>
            </div>
          </div>
        </div>
      </li>
      <li
        class="_tab_item_3abddc"
        title="label-test-3"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-test-3
              </span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`TabsMenu > should render with className 1`] = `
<div>
  <nav
    class="_tab_menu_3abddc className-test"
  >
    <ul />
  </nav>
</div>
`;

exports[`TabsMenu > should render with editable addable and removable 1`] = `
<div>
  <nav
    class="_tab_menu_3abddc"
  >
    <ul>
      <li
        class="_tab_item_3abddc _active_3abddc"
        title="label-test-1"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-test-1
              </span>
              <button
                class="_edit_button_732c04"
                title="Edit tag"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_2be8fc _primary_2be8fc _edit_icon_732c04"
                  fill="none"
                  viewBox="0 0 21 21"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.418 17.861 1 20l2.139-6.418m4.279 4.279 10.7-10.7a3.027 3.027 0 0 0-2.14-5.165c-.802 0-1.571.319-2.139.886l-10.7 10.7m4.279 4.279-4.279-4.279m2.139 2.14 7.844-7.844m-1.426-2.853 4.279 4.279"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <i>
                  Edit tag
                </i>
              </button>
            </div>
          </div>
        </div>
        <button
          class="_button_closer_8d9a3f _tab_closer_3abddc"
          title="Remove tab"
        >
          <svg
            aria-hidden="true"
            class="_icon_2be8fc _primary_2be8fc _icon_8d9a3f"
            fill="none"
            viewBox="-1 -1 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <i>
            Remove tab
          </i>
        </button>
      </li>
      <li
        class="_tab_item_3abddc"
        title="label-test-2"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-test-2
              </span>
              <button
                class="_edit_button_732c04"
                title="Edit tag"
              >
                <svg
                  aria-hidden="true"
                  class="_icon_2be8fc _primary_2be8fc _edit_icon_732c04"
                  fill="none"
                  viewBox="0 0 21 21"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M7.418 17.861 1 20l2.139-6.418m4.279 4.279 10.7-10.7a3.027 3.027 0 0 0-2.14-5.165c-.802 0-1.571.319-2.139.886l-10.7 10.7m4.279 4.279-4.279-4.279m2.139 2.14 7.844-7.844m-1.426-2.853 4.279 4.279"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
                <i>
                  Edit tag
                </i>
              </button>
            </div>
          </div>
        </div>
        <button
          class="_button_closer_8d9a3f _tab_closer_3abddc"
          title="Remove tab"
        >
          <svg
            aria-hidden="true"
            class="_icon_2be8fc _primary_2be8fc _icon_8d9a3f"
            fill="none"
            viewBox="-1 -1 16 16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <i>
            Remove tab
          </i>
        </button>
      </li>
      <li
        class="_add_tab_3abddc"
      >
        <a
          aria-label="Add new tab"
          href="#"
          title="Add new tab"
        >
          <svg
            aria-hidden="true"
            class="_icon_2be8fc _primary_2be8fc"
            fill="none"
            viewBox="-0.5 -0.5 19 19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 1v16M1 9h16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.5"
            />
          </svg>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`TabsMenu > should render with tab colors 1`] = `
<div>
  <nav
    class="_tab_menu_3abddc className-test"
  >
    <ul>
      <li
        class="_tab_item_3abddc"
        title="label-1"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <span
            class="_tab_color_3abddc"
            style="background: Red;"
          />
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-1
              </span>
            </div>
          </div>
        </div>
      </li>
      <li
        class="_tab_item_3abddc"
        title="label-2"
      >
        <div
          class="_tab_trigger_3abddc"
        >
          <span
            class="_tab_color_3abddc"
            style="background: Blue;"
          />
          <div
            class="_tab_editable_732c04 _tab_label_3abddc"
          >
            <div
              class="_label_732c04"
            >
              <span
                class="_text_732c04"
              >
                label-2
              </span>
            </div>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;
